src/main/scala/io/iohk/ethereum/vm/EvmConfig.scala

```diff
@@ -28,25 +28,26 @@ object EvmConfig {
     * returns the evm config that should be used for given block
     */
   def forBlock(blockNumber: BigInt, blockchainConfig: BlockchainConfigForEvm): EvmConfig = {
-    val transitionBlockToConfigMapping: Map[BigInt, EvmConfigBuilder] = Map(
-      blockchainConfig.frontierBlockNumber -> FrontierConfigBuilder,
-      blockchainConfig.homesteadBlockNumber -> HomesteadConfigBuilder,
-      blockchainConfig.eip150BlockNumber -> PostEIP150ConfigBuilder,
-      blockchainConfig.eip160BlockNumber -> PostEIP160ConfigBuilder,
-      blockchainConfig.eip161BlockNumber -> PostEIP161ConfigBuilder,
-      blockchainConfig.byzantiumBlockNumber -> ByzantiumConfigBuilder,
-      blockchainConfig.constantinopleBlockNumber -> ConstantinopleConfigBuilder,
-      blockchainConfig.istanbulBlockNumber -> IstanbulConfigBuilder,
-      blockchainConfig.atlantisBlockNumber -> AtlantisConfigBuilder,
-      blockchainConfig.aghartaBlockNumber -> AghartaConfigBuilder,
-      blockchainConfig.petersburgBlockNumber -> PetersburgConfigBuilder,
-      blockchainConfig.phoenixBlockNumber -> PhoenixConfigBuilder
+    val transitionBlockToConfigWithPriorityMapping: Map[BigInt, (Int, EvmConfigBuilder)] = Map(
+      blockchainConfig.frontierBlockNumber -> (1, FrontierConfigBuilder),
+      blockchainConfig.homesteadBlockNumber -> (2, HomesteadConfigBuilder),
+      blockchainConfig.eip150BlockNumber -> (3, PostEIP150ConfigBuilder),
+      blockchainConfig.eip160BlockNumber -> (4, PostEIP160ConfigBuilder),
+      blockchainConfig.eip161BlockNumber -> (5, PostEIP161ConfigBuilder),
+      blockchainConfig.byzantiumBlockNumber -> (6, ByzantiumConfigBuilder),
+      blockchainConfig.atlantisBlockNumber -> (6, AtlantisConfigBuilder),
```

Comment:
As i understand that it is also not 100% deterministic as in case of same block and priority the chooses fork will be random one.  Maybe leave some comment about it and create and leave some ticket to refactor our blockchain configs ? (we have probably been postponing it long enough)

Reply:
Not random but first checked. So in our test config, when we set all numbers to 0 expecting to use the newest but we are getting the first in the map (so it is the oldest fork)